Sources/Commands/Workspace.swift

```diff
@@ -287,6 +323,39 @@ public class Workspace {
         try removeFileTree(dataPath)
     }
 
+    /// Find the managed dependencies for the given package name.
+    func findDependency(forURL url: String) throws -> ManagedDependency? {
+        guard let dependency = dependencyMap[RepositorySpecifier(url: url)] else {
+            return nil
+        }
+        return dependency
+    }
+
+    /// Puts a dependency in edit mode creating a checkout in editables directory.
+    func edit(dependency: ManagedDependency, at revision: Revision, packageName: String) throws {
```

Comment:
Isn't `revision == dependency.currentRevision` here? If so, we could avoid passing it.